x/campaign/types/total_supply.go

```diff
@@ -21,3 +25,34 @@ func UpdateTotalSupply(coins, updatedCoins sdk.Coins) sdk.Coins {
 
 	return append(updatedCoins, notUpdated...).Sort()
 }
+
+// ValidateTotalSupply checks whether the total supply for each denom is within the provided total supply range
+func ValidateTotalSupply(coins sdk.Coins, supplyRange TotalSupplyRange) error {
+	// extra safety check, this should never happen
+	if supplyRange.MaxTotalSupply.LT(supplyRange.MinTotalSupply) {
```

Comment:
I think it would make sense to introduce a `func (TotalSupplyRange) Validate() error` method so the check is always performed from a single entrypoint
Then the param validate calls this method